src/main/java/reactor/netty/transport/ProxyProvider.java

```diff
@@ -161,18 +160,7 @@ public final ProxyHandler newProxyHandler() {
 	 * @return true if of type {@link InetSocketAddress} and hostname candidate to proxy
 	 */
 	public boolean shouldProxy(SocketAddress address) {
-		return address instanceof InetSocketAddress && shouldProxy(((InetSocketAddress) address).getHostString());
-	}
-
-	/**
-	 * Should proxy the given hostname
-	 *
-	 * @param hostName the hostname to test
-	 * @return true if candidate to proxy
-	 */
-	public boolean shouldProxy(@Nullable String hostName) {
-		return nonProxyHosts == null || hostName == null || !nonProxyHosts.matcher(hostName)
-		                                                                  .matches();
+		return address instanceof InetSocketAddress && shouldProxy.test(address);
```

Comment:
Let's clarify what's returning `shouldProxy` and then we can thing about backwards compatibility.